test/e2e/kubectl.go

```diff
@@ -97,11 +97,12 @@ var _ = Describe("kubectl", func() {
 	Describe("guestbook", func() {
 		var guestbookPath = filepath.Join(testContext.RepoRoot, "examples/guestbook")
 
-		if testContext.Provider != "gce" && testContext.Provider != "gke" {
-			By(fmt.Sprintf("Skipping guestbook, uses createExternalLoadBalancer, a (gce|gke) feature"))
-		}
-
 		It("should create and stop a working application", func() {
+			if testContext.Provider != "gce" && testContext.Provider != "gke" {
```

Comment:
Please add a check for testContext.Provider being empty string or nil, and Fail if it is, to prevent similar errors in the future.   We should ideally wrap this in a utility function in test/e2e/util.go so that other tests can benefit from it.